Hearthstone Deck Tracker/OpponentDeadForTracker.cs

```diff
@@ -0,0 +1,60 @@
+﻿using HearthDb.Enums;
+using Hearthstone_Deck_Tracker.Hearthstone;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Hearthstone_Deck_Tracker
+{
+	public static class OpponentDeadForTracker
+	{
+		private static List<string> _uniqueDeadHeroes = new List<string>();
+		private static List<int> _deadTracker = new List<int>();
+		const string KelThuzadCardId = "KelThuzad";
+
+		public static async void ShoppingStarted(GameV2 game)
+		{
+			if(game.GetTurnNumber() <= 1)
+			{
+				_uniqueDeadHeroes.Clear();
+				_deadTracker.Clear();
+				Core.Overlay.nextOpponentLeaderboardPosition = null;
+			}
+			for(int i =0; i < _deadTracker.Count; i++)
+			{
+				_deadTracker[i]++;
+			}
+			var deadHeroes = game.Entities.Select(x=>x.Value).Where(x => x.IsHero && x.Health <= 0);
+			foreach(var hero in deadHeroes)
+			{
+				var id = BattlegroundsBoardState.GetCorrectBoardstateHeroId(hero.CardId);
+				if(!id.Contains(KelThuzadCardId) && !_uniqueDeadHeroes.Contains(id))
+				{
+					_deadTracker.Add(0);
+					_uniqueDeadHeroes.Add(id);
+				}
+			}
+			_deadTracker.Sort((x, y) => y.CompareTo(x));
+			Core.Overlay.UpdateOpponentDeadForTurns(_deadTracker);
+			var gameEntites = game.Entities.Values;
+			var currentPlayer = gameEntites.FirstOrDefault(x => x.IsCurrentPlayer);
+			for(int i = 0; i < 5; i++)
+			{
+				if(currentPlayer != null && currentPlayer.HasTag(GameTag.NEXT_OPPONENT_PLAYER_ID))
+				{
+					var nextOpponent = gameEntites.FirstOrDefault(x => x.GetTag(GameTag.PLAYER_ID) == currentPlayer.GetTag(GameTag.NEXT_OPPONENT_PLAYER_ID));
+					if(nextOpponent != null)
+					{
+						var leaderboardPlace = nextOpponent.GetTag(GameTag.PLAYER_LEADERBOARD_PLACE);
+						if(leaderboardPlace >= 0 && leaderboardPlace < 8)
+						{
+							Core.Overlay.nextOpponentLeaderboardPosition = leaderboardPlace;
+							Core.Overlay.PositionDeadForText();
```

Comment:
I believe this is the success case? You can `break` the for loop here to avoid more iterations.

Reply:
We want to scan to the end of the loop because at the start of shopping that will be hit by the old opponent, so we want to wait until the very end to get the next opponent. Maybe this should just wait 2500 at the beginning and not loop; I added the loop thinking it would make it hit as soon as possible but might not be worth the trouble

Reply:
I see, makes sense. Adding that to the comment would be good as well.